_posts/2018-07-31-how-to-move-to-https.md

```diff
@@ -0,0 +1,100 @@
+
+- Hard-coding a resource: e.g. `http://www.example.com/img/insecure-image.jpg`
+- Using a 3rd-party library or ad network reference: `http://www.example.com/js/analytics.js`
+  - This is common for libraries that haven't been updated in a while. Most all of them now serve the same assets over the same path securely with HTTPS.
+
+Even [practices that were previously encouraged by Paul Irish](https://www.paulirish.com/2010/the-protocol-relative-url/), a leading web architect for the Google Chrome team may have contributed to your mixed content problem, so don't feel bad. But, there will likely be work to be done.
+
+#### The risk of mixed content
+
+These "not secure" bits of mixed-content [expose the same risk](https://developers.google.com/web/fundamentals/security/prevent-mixed-content/what-is-mixed-content#mixed_content_weakens_https
+) that your HTML does when served over HTTP, so browsers rightfully show the browser that the experience on your site is "not secure".
+
+Mixed content is categorized in two ways: active and passive. An asset that can contribute to passive mixed content would be an image; it doesn't interact with the page but is merely presented on the page. An example that can be an active asset of mixed content is a javascript file or stylesheet since its purpose is to manipulate the page. Passive mixed content, albeit categorically less severe than active, still exposes risk to an attacker to learn a lot about an individual's browsing patterns and even trick them into taking different actions than they intend to on the page. Therefore passive mixed content still constitutes enough of a threat for the browser to issue a warning display.
+
+![mixed content browser errors]({{ site.url }}/assets/img/blog/mixed-content-errors.png)
+
+In the case of an active asset that is compromised, if for example, it was a javascript file, an attacker can take full control of the page and any interaction you may have with it like entering passwords, or credit card information for example. The mechanism behind this is a [somewhat sophisticated man-in-the-middle attack](https://developers.google.com/web/fundamentals/security/prevent-mixed-content/what-is-mixed-content#mixed_content_types_security_threats_associated), but suffice it to say, if the browser recognizes the vulnerability, the [user experience we discussed in part one](/2018/06/26/why-you-need-https.html#https-improves-credibility-and-ux) will be poor in the best of scenarios, and total data compromise in the worst. Your audience, and by association your organization, will be seeing red.
+<br> ![Chrome misconfigured HTTPS]({{ site.url }}/assets/img/blog/misconfigured-https.png)
```

Comment:
"...if the browser recognizes the vulnerability, the best scenario is the poor user experience we discussed in part one, the worst is total data compromise."